GETTING_STARTED.md

```diff
@@ -0,0 +1,374 @@
++  return {
++    todos: store.todos.getAsArray(),
++  }
++})(Todos)
+
+const App = () => (
+-  <div>Hey!</div>
++  <TodosContainer /> // c.
+)
+
+export default provider(store)(App)
+```
+
+We create a classical React Component (a) and inject todos (as array) in its props (b), and finally we use this new component in our App (c)
+
+## 5. Add a new todos (store mutation)
+```diff
+import React from 'react'
+import { createStore, types } from 'k-ramel'
+import { provider, inject } from '@k-ramel/react'
+
+const store = createStore(
+  {
+    todos: types.keyValue(),
+  },
+)
+
+store.todos.add({ id: 2, title: 'finish the documentation' })
+store.todos.add({ id: 32, title: 'an other' })
+
+console.log(store.todos.get(2))
+
+-const Todos = ({ todos }) => (
++const Todos = ({ todos, onClick }) => (
++  <div>
++    <button onClick={onClick}>Add a todo!</button> {/* c. */}
+    <ul>
+      {todos.map(({ id, title }) => (
+        <li key={id}>{title}</li>
+      ))}
+    </ul>
++  </div>
+)
+
+const TodosContainer = inject((store) => {
+  return {
+    todos: store.todos.getAsArray(),
++    // a.
++    onClick: () => {
++      // b.
++      store.todos.add({ id: Math.random(), title: 'Yo! I am a new todo!' })
++    }
+  }
+})(Todos)
+
+const App = () => (
+  <TodosContainer />
+)
+
+export default provider(store)(App)
+```
+
+When we click on the new button (c), the function injected (a) is triggered. This function ask for a mutation (b), which add a new todo (the title is hardcoded here)
+
+## 6. Listen to the todo addition
+```diff
+import React from 'react'
+-import { createStore, types } from 'k-ramel'
++import { createStore, types, when } from 'k-ramel'
+-import { provider, inject } from '@k-ramel/react'
++import { provider, inject, listen } from '@k-ramel/react'
+
+const store = createStore(
+  {
+    todos: types.keyValue(),
+  },
+)
+
+store.todos.add({ id: 2, title: 'finish the documentation' })
+store.todos.add({ id: 32, title: 'an other' })
+
+console.log(store.todos.get(2))
+
+const Todos = ({ todos, onClick }) => (
+  <div>
+    <button onClick={onClick}>Add a todo!</button> {/* c. */}
+    <ul>
+      {todos.map(({ id, title }) => (
+        <li key={id}>{title}</li>
+      ))}
+    </ul>
+  </div>
+)
+
+const TodosContainer = inject((store) => {
+  return {
+    todos: store.todos.getAsArray(),
+    onClick: () => {
+      store.todos.add({ id: Math.random(), title: 'Yo! I am a new todo!' })
+    }
+  }
+})(Todos)
+
+const App = ({ onClick }) => (
+  <TodosContainer />
+)
++
++const listeners = [
++  when('@@krf/ADD>TODOS')(() => {
++    console.log('A todo is added!')
++  })
++]
++
++const AppContainer = listen(listeners)(App)
+
+-export default provider(store)(App)
++export default provider(store)(AppContainer)
+```
+
+Here we opened redux-devtools, click on the button, and look at actions that are logged into redux-devtools :).\
+We see that the action `@@krf>ADD>TODOS` is the one triggered when a new todo is added.\
```

Comment:
~We see that~ the action `@@krf>ADD>TODOS` is ~the one~ triggered when a new todo is added